src/llmcompressor/transformers/finetune/text_generation.py

```diff
@@ -199,16 +204,14 @@ def initialize_model_from_path(
         "trust_remote_code": model_args.trust_remote_code_model,
     }
     # this calls from_pretrained under the hood so should be FSDP safe
-    model = SparseAutoModel.text_generation_from_pretrained(
-        model_name_or_path=model_path,
-        sequence_length=None,  # use model default
+    model = AutoModelForCausalLM.from_pretrained(
+        model_path,
         **model_kwargs,
     )
 
     teacher = (
-        SparseAutoModel.text_generation_from_pretrained(
-            model_name_or_path=model_args.distill_teacher,
-            sequence_length=None,  # use model default
+        AutoModelForCausalLM.from_pretrained(
```

Comment:
Is there a need a to keep SparseAutoModel if we're swapping it to just use AutoModelForCausalLM?

Reply:
Its just for backwards compatibility, not needed for functionality.